From 054eb9804ef9513429f0c807b33497a91805a33a Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Sat, 2 Dec 2017 12:52:18 +0000
Subject: [PATCH 04/32] Settings: Add proximity check on wake preference

 * Splitted from commit 5aa2f34b1c78c5b41f25b820c1f29b29c58e0cae
   and adapted to the Lineage SDK.

Change-Id: I7516c833ff39de6bdb36c815a5292606cbaa0533
---
 res/values/cm_strings.xml                     | 4 ++++
 res/xml/display_settings.xml                  | 9 +++++++++
 src/com/android/settings/DisplaySettings.java | 5 +++++
 3 files changed, 18 insertions(+)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 83a69aeec0..da258f5b7e 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -29,4 +29,8 @@
     <!-- Double tap to sleep on status bar or lockscreen -->
     <string name="status_bar_double_tap_to_sleep_title">Tap to sleep</string>
     <string name="status_bar_double_tap_to_sleep_summary">Double-tap on the status bar or lockscreen to turn off the display</string>
+
+    <!-- Proximity wake -->
+    <string name="proximity_wake_title">Prevent accidental wake-up</string>
+    <string name="proximity_wake_summary">Check the proximity sensor prior to waking up screen</string>
 </resources>
diff --git a/res/xml/display_settings.xml b/res/xml/display_settings.xml
index ae4b063b11..342096e1e2 100644
--- a/res/xml/display_settings.xml
+++ b/res/xml/display_settings.xml
@@ -16,6 +16,7 @@
 
 <PreferenceScreen
     xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:lineage="http://schemas.android.com/apk/res/lineageos.platform"
     xmlns:settings="http://schemas.android.com/apk/res-auto"
     android:key="display_settings_screen"
     android:title="@string/display_settings"
@@ -136,6 +137,14 @@
         android:summary="@string/status_bar_double_tap_to_sleep_summary"
         android:defaultValue="true" />
 
+    <!-- Prevent accidental wake-up -->
+    <lineageos.preference.LineageSystemSettingSwitchPreference
+        android:key="proximity_on_wake"
+        android:title="@string/proximity_wake_title"
+        android:summary="@string/proximity_wake_summary"
+        android:defaultValue="false"
+        lineage:requiresConfig="@*lineageos.platform:bool/config_proximityCheckOnWake" />
+
     <ListPreference
         android:key="theme"
         android:title="@string/color_theme"
diff --git a/src/com/android/settings/DisplaySettings.java b/src/com/android/settings/DisplaySettings.java
index fef56ad582..2a83eae0e7 100644
--- a/src/com/android/settings/DisplaySettings.java
+++ b/src/com/android/settings/DisplaySettings.java
@@ -48,6 +48,7 @@ public class DisplaySettings extends DashboardFragment {
     private static final String TAG = "DisplaySettings";
 
     public static final String KEY_DISPLAY_SIZE = "display_settings_screen_zoom";
+    public static final String KEY_PROXIMITY_ON_WAKE = "proximity_on_wake";
 
     private static final String KEY_SCREEN_TIMEOUT = "screen_timeout";
     private static final String KEY_AMBIENT_DISPLAY = "ambient_display";
@@ -122,6 +123,10 @@ public class DisplaySettings extends DashboardFragment {
                     keys.add(WallpaperPreferenceController.KEY_WALLPAPER);
                     keys.add(KEY_NIGHT_DISPLAY);
                     keys.add(KEY_AUTO_BRIGHTNESS);
+                    if (!context.getResources().getBoolean(
+                            org.lineageos.platform.internal.R.bool.config_proximityCheckOnWake)) {
+                        keys.add(KEY_PROXIMITY_ON_WAKE);
+                    }
                     return keys;
                 }
 
-- 
2.17.1

